### DISCUSSION TITLE
Entities/Tables that only apply to one db when working with multiple

### DISCUSSION THREAD

**Discussion by dave08** [vincentlauvlwj/Ktorm#265] [Ideas]
I need to work with multiple databases, and their tables are different. Currently, all Entities/Tables can be mistakenly used by any database. I was thinking of one solution for us:
```kotlin
class DbOne(private val db: Database): Database by db {
    val oneTable get() = db.sequenceOf(TableOne)
}
```
but Database isn't an interface nor open, so it can't be derived from... are there any other suggested solutions to this problem?

By the way, this kind of possibility would also be useful for dependency injection... instead of using `@Named()` (in Micronaut, but also applies to other frameworks), we could inject the right db straight away.

**Comment by dmitchell** [comment]
We (Cargurus) use annotations `@Snowflake, @MySql, @Postgres` for this, but other than for auto-verifying `Table<>` definitions during integration testing, it's mostly for the developer. I'll have to think whether there's a reasonable place (e.g., on the code for `innerJoin` and its siblings) to ensure that all the tables have the same annotation.

We seldom use `Entity<>` via the `sequenceOf` (sometimes via `row.toEntity()` but not often as our `select`s are usually not all and just the columns of the `from(table)`)
